auth/auth.go

```diff
@@ -103,6 +103,9 @@ func CheckScrambledPassword(salt, hpwd, auth []byte) bool {
 	terror.Log(errors.Trace(err))
 	hash := crypt.Sum(nil)
 	// token = scrambleHash XOR stage1Hash
+	if len(auth) != len(hash) {
```

Comment:
Is it possible that `len(auth) > len(hash)`?

Reply:
It doesn't matter, as long as `len(auth) != len(hash)` the input is invalid.